pkg/zack/layout_border.go

```diff
@@ -0,0 +1,111 @@
+package zack
+
+import (
+	"context"
+
+	"github.com/charmbracelet/lipgloss"
+	"github.com/rs/zerolog/log"
+)
+
+var (
+	// DefaultBorderIdleStyle is the default style for the border when it is idle (not focused).
+	// It uses a normal border style with gray foreground color.
+	DefaultBorderIdleStyle = lipgloss.NewStyle().
+				BorderStyle(lipgloss.NormalBorder()).
+				BorderForeground(lipgloss.Color("240"))
+
+	// DefaultBorderActiveStyle is the default style for the border when it is active (focused).
+	// It uses a normal border style with blue foreground color.
+	DefaultBorderActiveStyle = DefaultBorderIdleStyle.
+					BorderForeground(lipgloss.Color("33"))
+)
+
+var _ Boundable = (*BorderLayout)(nil)
+
+// BorderLayout is a layout that renders a border around its content.
+//
+// +-------------------+
+// |                   |
+// |     content       |
+// |                   |
+// +-------------------+
```

Comment:
The 'content' text is not centered 😪
```suggestion
// +-------------------+
// |                   |
// |      content      |
// |                   |
// +-------------------+
```